### ISSUE TITLE
[base node] header sync stuck trying to sync from base node with no more RPC sessions available

### ISSUE THREAD

**Issue by hansieodendaal** [tari-project/tari#5078]
The base node keeps up with the blockchain, but the sync status gets stuck on syncing due to `header sync failed due to RPC session unavailability` without trying header sync from an alternative base node. This influences the base node status in the console wallet.

The events below are repeated over and over:

``` rust
2023-01-03 02:47:22.268102900 [c::bn::header_sync] DEBUG Starting header sync.
2023-01-03 02:47:22.268113300 [c::bn::header_sync] INFO  Synchronizing headers (1 candidate peers selected)
2023-01-03 02:47:22.268123200 [c::bn::header_sync] INFO  Attempting to sync headers (1 sync peers)
2023-01-03 02:47:22.268134400 [c::bn::header_sync] DEBUG Dialing 649a03a091df84ba537d1f7c72 sync peer
2023-01-03 02:47:22.268259300 [c::bn::header_sync] INFO  Successfully dialed sync peer 649a03a091df84ba537d1f7c72 in 124.60µs
2023-01-03 02:47:22.268271200 [c::bn::header_sync] DEBUG Attempting to synchronize headers with `649a03a091df84ba537d1f7c72`
2023-01-03 02:47:23.795555300 [c::bn::header_sync] DEBUG Header sync failed: RPC error: Handshake error: RPC handshake was explicitly rejected: no more RPC sessions available
.
.
.
2023-01-03 05:45:28.594879300 [c::bn::header_sync] DEBUG Starting header sync.
2023-01-03 05:45:28.594882500 [c::bn::header_sync] INFO  Synchronizing headers (1 candidate peers selected)
2023-01-03 05:45:28.594886300 [c::bn::header_sync] INFO  Attempting to sync headers (1 sync peers)
2023-01-03 05:45:28.594890500 [c::bn::header_sync] DEBUG Dialing 649a03a091df84ba537d1f7c72 sync peer
2023-01-03 05:45:28.594937600 [c::bn::header_sync] INFO  Successfully dialed sync peer 649a03a091df84ba537d1f7c72 in 47.00µs
2023-01-03 05:45:28.594941700 [c::bn::header_sync] DEBUG Attempting to synchronize headers with `649a03a091df84ba537d1f7c72`
2023-01-03 05:45:29.483111400 [c::bn::header_sync] DEBUG Header sync failed: RPC error: Handshake error: RPC handshake was explicitly rejected: no more RPC sessions available```
```

Network log:
``` rust
2023-01-03 05:45:28.594900600 [comms::connectivity::manager] TRACE Request: DialPeer { node_id: NodeId(649a03a091df84ba537d1f7c72), reply_tx: Some(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: true, is_tx_task_set: false } }) }) }
2023-01-03 05:45:28.594929700 [comms::connectivity::manager] DEBUG Found existing connection for peer `649a03a091df84ba`
2023-01-03 05:45:28.594946600 [comms::connection_manager::peer_connection] DEBUG Attempting to establish RPC protocol `t/blksync/1` to peer `649a03a091df84ba537d1f7c72`
2023-01-03 05:45:28.594957100 [tari_comms::connection_manager::peer_connection] INFO  open_negotiated_protocol_stream; protocol=b"t/blksync/1"
2023-01-03 05:45:28.594962200 [comms::connection_manager::peer_connection] DEBUG [PeerConnection(id=5615, peer_node_id=649a03a091df84ba, direction=Outbound)] Negotiating protocol 't/blksync/1' on new substream for peer '649a03a091df84ba'
2023-01-03 05:45:28.595025900 [comms::connection_manager::protocol] TRACE Wrote frame 't/blksync/1' (11 byte(s) Flags=OPTIMISTIC | TERMINATE)
2023-01-03 05:45:28.595046500 [comms::rpc::client] DEBUG (stream=671) Performing client handshake for 't/blksync/1'
2023-01-03 05:45:28.595052100 [tari_comms::protocol::rpc::handshake] INFO  rpc::client::perform_client_handshake; comms.direction="outbound"
2023-01-03 05:45:28.595058200 [comms::rpc::handshake] DEBUG Sending client handshake (3 bytes)
.
.
.
2023-01-03 05:45:29.483053800 [tari_comms::protocol::rpc::handshake] ERROR error=RPC handshake was explicitly rejected: no more RPC sessions available

```

Console output:

``` rust
05:26 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 40 Rpc: 86/10000 ⚡️ 792.35ms
05:26 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 41 Rpc: 86/10000 ⚡️ 1.27s
05:26 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 41 Rpc: 86/10000 ⚡️ 1.27s
05:26 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 1.27s
05:26 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 41 Rpc: 86/10000 ⚡️ 1.27s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 40 Rpc: 86/10000 ⚡️ 722.05ms
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 38 Rpc: 86/10000 ⚡️ 722.05ms
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14136 (Tue, 03 Jan 2023 03:26:26 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 37 Rpc: 86/10000 ⚡️ 722.05ms
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 52 Rpc: 86/10000 ⚡️ 1.45s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 50 Rpc: 86/10000 ⚡️ 1.45s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 38 Rpc: 86/10000 ⚡️ 1.45s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 38 Rpc: 86/10000 ⚡️ 2.31s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 37 Rpc: 86/10000 ⚡️ 2.31s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 37 Rpc: 86/10000 ⚡️ 2.31s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 36 Rpc: 86/10000 ⚡️ 1.85s
05:27 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 36 Rpc: 86/10000 ⚡️ 1.85s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 37 Rpc: 86/10000 ⚡️ 1.85s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 39 Rpc: 86/10000 ⚡️ 1.82s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 39 Rpc: 86/10000 ⚡️ 1.82s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 39 Rpc: 86/10000 ⚡️ 1.82s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 26 Rpc: 86/10000 ⚡️ 1.82s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14137 (Tue, 03 Jan 2023 03:27:14 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 28 Rpc: 86/10000 ⚡️ 2.39s
05:28 v0.43.1 esmeralda State: Listening Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 40 Rpc: 86/10000 ⚡️ 2.39s RandomX: #5 with flags FLAG_HARD_AES | FLAG_JIT | FLAG_ARGON2_SSSE3 | FLAG_ARGON2_AVX2 | FLAG_ARGON2
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 2.39s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 43 Rpc: 86/10000 ⚡️ 2.72s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 43 Rpc: 86/10000 ⚡️ 2.72s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 2.72s
05:28 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 2.72s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 1.70s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 41 Rpc: 86/10000 ⚡️ 1.70s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 1.70s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 42 Rpc: 86/10000 ⚡️ 2.74s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 41 Rpc: 86/10000 ⚡️ 2.74s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 40 Rpc: 86/10000 ⚡️ 2.74s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 28 Rpc: 86/10000 ⚡️ 2.74s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 26 Rpc: 86/10000 ⚡️ 2.62s
05:29 v0.43.1 esmeralda State: Sync failed: HeaderSyncFailed Tip: 14138 (Tue, 03 Jan 2023 03:27:46 +0000) Mempool: 0tx (0g, +/- 0blks) Connections: 14|16 Banned: 1 Messages (last 60s): 25 Rpc: 86/10000 ⚡️ 2.62s
```

**Comment by sdbondi** [comment]
Thanks for the issue, please confirm that force_sync_peers was not set. If it is not, then there is an issue with the peers that were provided to the header_sync component / the header_sync component is not switching other peers on failure. 

Primary: discover the bug either in the listening state or in the header sync itself and fix it.
Alternate:
Banning may "fix" this but doesn't seem appropriate given the peer is probably not malicious and bans decrease overall network effectiveness. An idea for these circumstances is to maintain a list of temporarily faulty peers and ignore their chain metadata for an extended period while still remaining connected. A short ban is an option (not opposed), but would only "fix" the problem for the short length of the ban.

**Comment by SWvheerden** [comment]
mmm maybe a short ban here should be added

**Comment by SWvheerden** [comment]
closed by #5080

**Comment by hansieodendaal** [comment]
> Thanks for the issue, please confirm that `force_sync_peers` was not set. > 

Confirmed,  `force_sync_peers` was not set
